ZENKO-3643: added a refresh button for the data browser page

JulienMeziere opened the pull request:
I created a ListLayoutButtons and a ButtonsContainer to prepare for the future buttons that will have to be implemented according to the ticket's screenshot:

![image](https://user-images.githubusercontent.com/22881801/138286480-f58e08b2-ba4b-4a1b-9d01-1c1cfdd7f886.png)

JIRA ticket: https://scality.atlassian.net/browse/ZENKO-3643

Commit:
ZENKO-3643: added a refresh button for the data browser page

bert-e commented:
# Hello julienmeziere,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect Jira project



The Jira issue ZENKO-3643 specified in the source
branch name, does not belong to project `ZKUI`.

naren-rajendran commented on src/react/ui-elements/ListLayoutButtons.jsx:
```diff
@@ -0,0 +1,45 @@
+    const dispatch = useDispatch();
+    const isBrowsingObjects = !!matchPath(pathname, '/buckets/:bucketName/objects');
+    const prefixWithSlash = useMemo(() => {
+        const splitedPath = pathname.split('objects/');
```
small change, `splitedPath` -> `splittedPath`

JulienMeziere replied:
Should be good now 👍️

Commit:
Fix variable mispelling: changed 'splited' to 'splitted'

Cuervino reviewed (approved):
Approved (following a live demo)

nicolas2bert commented on src/react/ui-elements/ListLayoutButtons.jsx:
```diff
@@ -0,0 +1,45 @@
+import styled from 'styled-components';
+import { useDispatch } from 'react-redux';
+
+export const ButtonsContainer = styled.div`
```
Can this be moved to ui-elements/Container.jsx ?

JulienMeziere replied:
Should be good now 👍️

nicolas2bert commented on src/react/ui-elements/ListLayoutButtons.jsx:
```diff
@@ -0,0 +1,45 @@
+        if (splittedPath.length < 2 || splittedPath[1].length === 0) {
+            return '';
+        } else {
+            const prefix = splittedPath[1].slice(-1) === '/' ? splittedPath[1] : `${splittedPath[1]}/`;
```
You can use the `addTrailingSlash` utility method.

JulienMeziere replied:
Should be good now 👍️

nicolas2bert commented on src/react/ui-elements/ListLayoutButtons.jsx:
```diff
@@ -0,0 +1,45 @@
+import React, { useMemo } from 'react';
```
This `Buttons` Redux container is not really an ui component and is very specific to data browser, so we should move it to `databrowser` directory.

JulienMeziere replied:
I moved it in HeaderButtons inside the databrowser folder, is it good like this?

Commit:
Moved the header buttons specific to the data browser in the databrowser/ folder
Moved the ButtonsContainer component in ui-elements/Container.jsx
Used the addTrailingSlash method from the utilities

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **bypass_jira_check**

bert-e commented:
# Incorrect Jira project



The Jira issue ZENKO-3643 specified in the source
branch name, does not belong to project `ZKUI`.

elenamarcova commented:
/bypass_jira_check

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **bypass_jira_check**

nicolas2bert reviewed (approved):
@Cuervino do you mind doing a last check since couple changes have been made? Thank you :)

Cuervino commented:
> @Cuervino do you mind doing a last check since couple changes have been made? Thank you :)

Sure, I'll so it!

bert-e commented:
# Conflict




There is a conflict between your branch `feature/ZENKO-3643/add-a-refresh-button-for-the-data-browser` and the
destination branch `development/1.0`.

Please resolve the conflict on **the feature branch** (`feature/ZENKO-3643/add-a-refresh-button-for-the-data-browser`).

```sh
git fetch && \
git checkout origin/feature/ZENKO-3643/add-a-refresh-button-for-the-data-browser && \
git merge origin/development/1.0
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:feature/ZENKO-3643/add-a-refresh-button-for-the-data-browser
```







*The following options are set:* **bypass_jira_check**

Cuervino reviewed (approved):
lgtm

bert-e commented:
# Not Author



I'm afraid I cannot do that, @cuervino:

Only the author of the pull request @julienmeziere can use this command.

Please **delete** the corresponding comment so I can move on.




*The following options are set:* **bypass_jira_check**

bert-e commented:
# Conflict




There is a conflict between your branch `feature/ZENKO-3643/add-a-refresh-button-for-the-data-browser` and the
destination branch `development/1.0`.

Please resolve the conflict on **the feature branch** (`feature/ZENKO-3643/add-a-refresh-button-for-the-data-browser`).

```sh
git fetch && \
git checkout origin/feature/ZENKO-3643/add-a-refresh-button-for-the-data-browser && \
git merge origin/development/1.0
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:feature/ZENKO-3643/add-a-refresh-button-for-the-data-browser
```







*The following options are set:* **bypass_jira_check**

Commit:
Merge remote-tracking branch 'origin/development/1.0' into HEAD

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **bypass_jira_check**

JulienMeziere commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **bypass_jira_check, approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue ZENKO-3643.

Goodbye julienmeziere.
